nautobot/extras/tests/test_views.py

```diff
@@ -909,3 +912,104 @@ def test_create_object_with_constrained_permission(self):
         for cf in CustomField.objects.all():
             cf.delete()
         super().test_create_object_with_constrained_permission()
+
+
+class ScheduledJobTestCase(
+    ViewTestCases.ListObjectsViewTestCase,
+):
+    model = ScheduledJob
+
+    @classmethod
+    def setUpTestData(cls):
+        user = User.objects.create(username="user1", is_active=True)
+        ScheduledJob.objects.create(
+            name="test1",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="-",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            start_time=datetime.now(),
+        )
+        ScheduledJob.objects.create(
+            name="test2",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="-",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            start_time=datetime.now(),
+        )
+        ScheduledJob.objects.create(
+            name="test3",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="-",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            start_time=datetime.now(),
+        )
+
+
+class ApprovalQueueTestCase(
+    ViewTestCases.ListObjectsViewTestCase,
+):
+    model = ScheduledJob
+
+    def _get_url(self, action, instance=None):
+        if action != "list":
+            raise ValueError("This override is only valid for list test cases")
+        return reverse("extras:scheduledjob_approval_queue_list")
+
+    @classmethod
+    def setUpTestData(cls):
+        user = User.objects.create(username="user1", is_active=True)
+        ScheduledJob.objects.create(
+            name="test1",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="-",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            approval_required=True,
+            start_time=datetime.now(),
+        )
+        ScheduledJob.objects.create(
+            name="test2",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="-",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            approval_required=True,
+            start_time=datetime.now(),
+        )
+        ScheduledJob.objects.create(
+            name="test3",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="-",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            approval_required=True,
+            start_time=datetime.now(),
+        )
+
```

Comment:
Might be good to add a custom test function here to verify that only jobs needing approval are included in the listing?